app/src/androidTest/java/com/github/fribourgsdp/radio/LyricsGetterTest.kt

```diff
@@ -0,0 +1,131 @@
+package com.github.fribourgsdp.radio
+
+import okhttp3.*
+import org.junit.Test
+import org.junit.Assert.*
+import java.io.IOException
+import java.util.concurrent.ExecutionException
+
+
+/**
+ * Lyrics Getter Test
+ */
+//@RunWith(AndroidJUnit4::class)
+class LyricsGetterTest {
+    @Test
+    fun getLyricsFromSongAndArtist(){
+        val f = LyricsGetter.getLyrics("rouge", "sardou", OkHttpClient())
+        val lyrics = f.get()
+        assertTrue(lyrics.startsWith("Rouge\nComme un soleil couchant de Méditerranée"))
+    }
+    @Test
+    fun emptyLyricsTest(){
+        val lyricsFuture = LyricsGetter.getLyrics("stream", "dream theater", OkHttpClient())
+        val lyrics : String = lyricsFuture.get()
+        assertTrue(lyrics.equals("---No lyrics were found for this song.---"))
+    }
+    @Test(expected = Exception::class)
+    fun songIDsongNotFoundTest(){
+        val songIDFuture = LyricsGetter.getSongID("fsdgfdgdfgdfgdfg", "weoir hpfasdsfno", OkHttpClient())
+        songIDFuture.get()
+    }
+    @Test
+    fun songNotFoundTest(){
+        val lyricsFuture = LyricsGetter.getLyrics("fsdgfdgdfgdfgdfg", "weoir hpfasdsfno", OkHttpClient())
+        val lyrics = lyricsFuture.get()
+        assertTrue(lyrics.equals("---No lyrics were found for this song.---"))
+    }
+    @Test(expected = ExecutionException::class)
+    fun getLyricsFailingHTTPClientTest(){
+        val f = LyricsGetter.getLyrics("rouge", "sardou", SemiFailingHTTPClient())
+        val lyrics = f.get()
+        println(lyrics)
```

Comment:
Should this print and the one on line 48 stay in the tests?

Reply:
I thought they were useful to guarantee that lines 41 and 47 get executed. If there is no side-effect, they are considered as Dead-code. Do you suggest a cleaner way of doing this ?

Reply:
Hmm I see. Maybe do something arbitrary like concatenate those lyrics to another string, just to avoid unnecessary pollution in the console.